rate_limiting.go

```diff
@@ -0,0 +1,194 @@
+package cloudflare
+
+import (
+	"encoding/json"
+	"github.com/pkg/errors"
```

Comment:
Nit: the style elsewhere in this project (presumably because most people are using `goimports`) is to put external imports at the end, after a blank line.